File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;


class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> participantMap = new HashMap<>();
        
        for (String part : participant) {
            participantMap.put(part, participantMap.getOrDefault(part, 0) + 1);

        }
        
        for (String comp : completion) {
            participantMap.put(comp, participantMap.getOrDefault(comp, 0) - 1);
        }
        
        for (Map.Entry<String, Integer> entry : participantMap.entrySet()){
            if (entry.getValue() > 0){
                return entry.getKey();
            }
        }
        
        return "";
        
        
        
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    counts = defaultdict(int)
    for name in participant:
        counts[name] = counts[name] + 1

    for name in completion:
        counts[name] -= 1

    for name, cnt in counts.items():
        if cnt > 0:
            return name

